src/sdf/interval.h

```diff
@@ -0,0 +1,207 @@
+// Copyright 2024 The Manifold Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#pragma once
+
+#include <algorithm>
+#include <limits>
+
+#include "manifold/common.h"
+
+namespace manifold::sdf {
+
+// not really a precise implementation...
+template <typename Domain>
+struct Interval {
```

Comment:
This is a pretty general interval arithmetic implementation that covers common math functions. I tried to make things as precise as possible, but things like mod, for example, is hard to make precise and probably subject to changes later.